zeroex/ordervalidate/order_validator.go

```diff
@@ -808,33 +809,33 @@ func (o *OrderValidator) computeOptimalChunkSizes(signedOrders []*SignedOrder) [
 }
 
 func isSupportedSignature(signature []byte, orderHash common.Hash) bool {
-	signatureType := SignatureType(signature[len(signature)-1])
+	signatureType := zeroex.SignatureType(signature[len(signature)-1])
 
 	switch signatureType {
-	case IllegalSignature:
-	case InvalidSignature:
+	case zeroex.IllegalSignature:
```

Comment:
Sorry I didn't notice this before. In Go, switch cases do not implicitly fall through. As it is currently written `case zeroex.IllegalSignature` doesn't have any associated statements which means "break the switch statement and do nothing". You probably want to write it this way:

```go
case zeroex.IllegalSignature, zeroex.InvalidSignature:
    return false
```

See https://tour.golang.org/flowcontrol/9.

We can address this in a separate PR if you prefer.

Reply:
Thanks for catching. Since it did not impact the actual logic, I decided to fix it here.

Reply:
This *does* impact the actual logic. In the old code, the `IllegalSignature` case would break the switch and cause the function to return true. Maybe you're saying that it ultimately didn't affect anything because the signature is redundantly validated by the smart contract?

Reply:
Yes, that's what I meant. It didn't actually cause the validation to misbehave from an end-users perspective.